File: data/src/main/java/com.data/schedulers/SchedulersModule.java
package com.data.schedulers;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class SchedulersModule {

    @Provides
    @Singleton
    SchedulerProvider schedulerProvider() {
        return new SchedulersProviderImpl();
    }
}

File: commons/src/main/java/org/dhis2/commons/schedulers/SchedulersModule.kt
package org.dhis2.commons.schedulers

import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class SchedulersModule {
    @Provides
    @Singleton
    fun schedulerProvider(): SchedulerProvider {
        return SchedulersProviderImpl()
    }
}
